src/main/java/org/jitsi/videobridge/AbstractEndpointMessageTransport.java

```diff
@@ -417,9 +421,12 @@ protected void onSelectedEndpointsChangedEvent(
             videoConstraintsCompatibility.computeVideoConstraints());
     }
 
-    protected void onVideoConstraintsChangedEvent(Map<String, VideoConstraints> newVideoConstraints)
+    protected void onVideoConstraintsChangedEvent(Map<String, VideoConstraints> videoConstraintMap)
     {
-        endpoint.setVideoConstraints(newVideoConstraints);
+        // Don't "pollute" the video constraints map with constraints for this
+        // endpoint.
+        videoConstraintMap.remove(endpoint.getID());
+        endpoint.setSenderVideoConstraints(ImmutableMap.copyOf(videoConstraintMap));
```

Comment:
We modify the map on the line above, so this code must own the map. If that's the case, why is the copy necessary?

Reply:
> We modify the map on the line above, so this code must own the map.

That is correct.

> If that's the case, why is the copy necessary?

Not necessary strictly speaking, I just wanted to express in the API that the map is immutable and that requires a new object. We could go with normal maps if you'd prefer that.